sdk/core/Azure.Core/src/ClientUtilities.cs

```diff
@@ -201,14 +174,10 @@ public static void AssertEnumDefined(Type enumType, object value, string name)
         /// <param name="value">The value to validate.</param>
         /// <param name="name">The name of the parameter.</param>
         /// <exception cref="ArgumentNullException"><paramref name="value"/> has not been initialized.</exception>
-#if AZURE_NULLABLE
-        public static T CheckNotNull<T>([AllowNull, NotNull] T value, string name) where T : class
-#else
-        public static T CheckNotNull<T>(T value, string name) where T : class
-#endif
+        public static T CheckNotNull<T>(T? value, string name) where T : class
```

Comment:
Need to investigate where this is used and why AssertNotNull isn't sufficent there?